File: sentimental-credit/credit.py
from cs50 import get_int


def main():
    while True:
        num = get_int("Number: ")
        if (num > 0):
            break
    card_type(num)


def card_type(num):
    sumTwos = 0
    sumOnes = 0
    i = num // 10

    while (i > 0):
        sumTwos += ((i % 10) * 2)//10 + ((i % 10) * 2) % 10
        i //= 100

    i = num
    while (i > 0):
        sumOnes += i % 10
        i //= 100

    if (sumOnes + sumTwos) % 10 == 0:
        print(card_brand(num))
    else:
        print("INVALID\n")


def card_brand(num):
    digits = len(str(num))

    if digits in [13, 16] and (num // 10 ** 12 == 4 or num // 10 ** 15 == 4):
        return "VISA\n"
    elif digits == 16 and (num // 100000000000000 in [51, 52, 53, 54, 55]):
        return "MASTERCARD\n"
    elif digits == 15 and (num // 10000000000000 in [34, 37]):
        return "AMEX\n"
    return "INVALID\n"


main()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>


string getCardType(long cardNum);
int main(void)
{
   long cardNum;
    do{
        cardNum = get_long("Card Number: ");
    }
    while(cardNum < 0);

   printf("%s", getCardType(cardNum));



}

string getCardType(long cardNum){
    int sumOfTwos = 0;

    for(long i = cardNum /10; i > 0; i /= 100){

        sumOfTwos += ((i % 10) * 2)/10 + ((i % 10) * 2)%10;
    }

    int sumOfOnes = 0;

    for(long i = cardNum; i > 0; i /= 100){
        sumOfOnes += i % 10;
    }


   if((sumOfOnes + sumOfTwos)% 10 == 0){
      int digits = 0;

      for(long i = cardNum; i > 0; i /=10){
         digits++;
      }
      //VISA                                                                          4012888888881881
      if((digits == 13 || digits == 16) && (cardNum / 1000000000000 == 4 || cardNum / 1000000000000000 == 4)){
         return "VISA\n";
      }
      //MASTERCARD
      else if(digits == 16 && cardNum / 100000000000000 >= 51 && cardNum / 100000000000000 <= 55){
         return "MASTERCARD\n";
      }
      //AMEX                             378282246310005                   378282246310005
      else if(digits == 15 && (cardNum / 10000000000000 == 34 || cardNum / 10000000000000 == 37)){
         return "AMEX\n";
      }
      else{
         return "INVALID\n";
      }
   }
   else{
      return "INVALID\n";
   }

}


